### PR TITLE
pre-commit: integrate with cndp

### PR DESCRIPTION
**Pull request by maryamtahhan**
Install pre-commit for CNDP + its configuration.
Remove old githook.

### CODE REVIEW

**Comment by maryamtahhan** [comment]
Sorry for the delay in this, I was away on some summer hols - fixed up the last outstanding issue and it's ready now

File: lang/rs/wireguard/patch/0001-Integrate-CNDP-AF-XDP-with-Rust-Wireguard.patch [hunk 1]
```diff
@@ -91,7 +91,7 @@ index 640d8cb..7cb769b 100644
 @@ -5,12 +5,22 @@ license = "MIT"
  name = "wireguard-rs"
  version = "0.1.4"
- 
+
 +[[bin]]
```
Comment (KeithWiles) [inline lang/rs/wireguard/patch/0001-Integrate-CNDP-AF-XDP-with-Rust-Wireguard.patch:95]: Changing the patch file could be a problem here, need to verify or we should ignore this file.

File: lang/rs/wireguard/patch/0003-Update-latest-crypto-library-version.patch [hunk 2]
```diff
@@ -32,15 +32,15 @@ index 2941071..51cd887 100644
  dependencies = [
   "gimli",
  ]
- 
+
  [[package]]
```
Comment (KeithWiles) [inline lang/rs/wireguard/patch/0003-Update-latest-crypto-library-version.patch:36]: Same problem for this patch file.

**Review by KeithWiles — APPROVED** [review]
Going to approve, but I have a couple comments we may need to address.

We had a lot of mis-spelled words, which I most likely added :-)

This is a lot of work, thank you very much!

File: lang/rs/wireguard/patch/0001-Integrate-CNDP-AF-XDP-with-Rust-Wireguard.patch (hunk 1 above)
Comment (maryamtahhan) [reply lang/rs/wireguard/patch/0001-Integrate-CNDP-AF-XDP-with-Rust-Wireguard.patch:95]: Yeah, let me reset these and add them to the ignored files... sorry

**Pull request merged**